src/main/scala/com/fasterxml/jackson/module/scala/modifiers/ScalaTypeModifier.scala

```diff
@@ -0,0 +1,33 @@
+package com.fasterxml.jackson.module.scala.modifiers
+
+import java.lang.reflect.{ParameterizedType, Type}
+
+import com.fasterxml.jackson.databind.JavaType
+import com.fasterxml.jackson.databind.`type`._
+import com.fasterxml.jackson.module.scala.JacksonModule
+
+class ScalaTypeModifier extends TypeModifier {
+  override def modifyType(javaType: JavaType,
+                          jdkType: Type,
+                          context: TypeBindings,
+                          typeFactory: TypeFactory): JavaType = {
+
+
+    if (javaType.isTypeOrSubTypeOf(classOf[collection.Map[_, _]])) {
+      MapLikeType.upgradeFrom(javaType, javaType.containedTypeOrUnknown(0), javaType.containedTypeOrUnknown(1))
+    } else if (javaType.isTypeOrSubTypeOf(classOf[collection.TraversableOnce[_]])) {
+      CollectionLikeType.upgradeFrom(javaType, javaType.containedTypeOrUnknown(0))
+    } else if (javaType.isTypeOrSubTypeOf(classOf[Option[_]])) {
+      ReferenceType.upgradeFrom(javaType, javaType.containedTypeOrUnknown(0))
```

Comment:
Yes. As long as `Some` (and its friend(s)... `Either`?) implement or extend another `ReferenceType`, they will be considered `ReferenceType`s as well. Same is true for map/map-like and collection/collection-like types as well.